src/lecture_feedback/thread_safe_dict.py

```diff
@@ -1,20 +1,22 @@
+"""A simple thread-safe dictionary implementation."""
+
 import threading
 from collections.abc import Iterator
 from typing import Any
 
 
 class ThreadSafeDict:
-    """A simple thread-safe dictionary with only the methods we need"""
+    """A simple thread-safe dictionary with only the methods we need."""
 
     def __init__(self) -> None:
         self._data: dict[str, Any] = {}
         self._lock = threading.RLock()
 
-    def __setitem__(self, key: str, value: Any) -> None:
+    def __setitem__(self, key: str, value: Any) -> None:  # noqa: ANN401
```

Comment:
```suggestion
    def __setitem__(self, key: str, value: Any) -> None:  # noqa: ANN401 (any-type)
```

Reply:
Isn't this redundant? Is this just for your understanding?

Reply:
I dont know what ANN401 is so i would need to look it up, in contrast "(any-type)" is more human readable.
but its duplication up to you
[optional]

Reply:
In a an IDE with the ruff plugin (e.g., VS Code), you just need to hover over the code and it'll show you the check. Thus, I'll not go with this